src/Kernel/ClassDescription.class.st

```diff
@@ -959,6 +979,9 @@ ClassDescription >> renameProtocol: anOldProtocol as: aNewProtocol [
 	(aNewProtocol isNil or: [ anOldProtocol isNil ]) ifTrue: [ ^ self ].
 
 	(self hasProtocol: anOldProtocol) ifFalse: [ ^ self ].
+	
+	"We should not be able to have a protocol that is an extension of the same class so let's skip the rename."
+	(self isProtocolExtensionFromTheSamePackage: aNewProtocol) ifTrue: [ ^ self ].
```

Comment:
The same as before, it is better to generate a warning.

Reply:
Having something that does nothing without telling the user is really bad behaviour.